entity_api/src/lib.rs

```diff
@@ -16,6 +17,13 @@ pub(crate) fn uuid_parse_str(uuid_str: &str) -> Result<Id, error::Error> {
     })
 }
 
+pub(crate) fn naive_date_parse_str(date_str: &str) -> Result<chrono::NaiveDate, error::Error> {
+    chrono::NaiveDate::parse_from_str(date_str, "%Y-%m-%d").map_err(|_| error::Error {
```

Comment:
I'm curious if there's a specific reason you chose `NaiveDate` over `DateTime`? It seems that if you were to use `DateTime` instead, you would have a [more flexible string parser](https://docs.rs/chrono/latest/chrono/struct.DateTime.html#method.parse_from_rfc3339) to call, and YYY-MM-DD is a valid ISO8601 date. The reason I noticed this is because I was working with the same date format on the frontend and was able to use the `ts-luxmon` library's `DateTime` type to parse ISO8601 dates.

Reply:
Although, I did just notice you're only using this locally for tests, so perhaps this doesn't matter. Your call on this.

Reply:
I chose `NaiveDate` because `DateTime` is "timezone aware" as stated in their [docs](https://docs.rs/chrono/latest/chrono/). In our case we chose to set the timezone as a separate column on the table as opposed to encoded in the actual timestamps. We should definitely come back and revisit how this is set up and make sure it's what we're looking for. I'd propose considering that worthy of a discussion and a separate PR if that works for you.

Reply:
Ok sounds good. I created a [discussion to bookmark this conversation](https://github.com/Jim-Hodapp-Coaching/refactor-platform-rs/discussions/51#discussion-6901596) and discuss at the right time.